spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ReactiveKafkaAutoConfiguration.java

```diff
@@ -0,0 +1,98 @@
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(ReceiverOptions.class)
+	public <K, V> ReceiverOptions<K, V> receiverOptions() {
+		KafkaProperties.Reactor.Receiver receiverProperties = this.kafkaProperties.getReactor().getReceiver();
+		MutuallyExclusiveConfigurationPropertiesException.throwIfMultipleNonNullValuesIn((entries) -> {
+			entries.put("spring.kafka.reactor.receiver.subscribe-topics", receiverProperties.getSubscribeTopics());
+			entries.put("spring.kafka.reactor.receiver.subscribe-pattern", receiverProperties.getSubscribePattern());
+		});
+		ReceiverOptions<K, V> receiverOptions = ReceiverOptions.create(this.kafkaProperties.buildReactorConsumerProperties());
+
+		PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getCloseTimeout), receiverOptions::closeTimeout, receiverOptions);
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getPollTimeout), receiverOptions::pollTimeout, receiverOptions);
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getCommitInterval), receiverOptions::commitInterval, receiverOptions);
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getSubscribeTopics), receiverOptions::subscription, receiverOptions);
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getSubscribePattern), receiverOptions::subscription, receiverOptions);
+
+		map = map.alwaysApplying(this::whenPositveInt);
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getAtmostOnceCommitAheadSize), receiverOptions::atmostOnceCommitAheadSize, receiverOptions);
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getMaxDeferredCommits), receiverOptions::maxDeferredCommits, receiverOptions);
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getMaxCommitAttempts), receiverOptions::maxCommitAttempts, receiverOptions);
+		receiverOptions = toInstanceSafe(map.from(receiverProperties::getCommitBatchSize), receiverOptions::commitBatchSize, receiverOptions);
+		return receiverOptions;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(SenderOptions.class)
+	public <K, V> SenderOptions<K, V> senderOptions() {
+		SenderOptions<K, V> senderOptions = SenderOptions.create(this.kafkaProperties.buildReactorProducerProperties());
+		KafkaProperties.Reactor.Sender senderProperties = this.kafkaProperties.getReactor().getSender();
+		PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
+		senderOptions = toInstanceSafe(map.from(senderProperties::getCloseTimeout), senderOptions::closeTimeout, senderOptions);
+		senderOptions = toInstanceSafe(map.from(senderProperties::getMaxInFlight), senderOptions::maxInFlight, senderOptions);
+		senderOptions = toInstanceSafe(map.from(senderProperties::isStopOnError), senderOptions::stopOnError, senderOptions);
+		return senderOptions;
+	}
+
+	private <T, R> R toInstanceSafe(PropertyMapper.Source<T> source, Function<T, R> target, R defaultOptions) {
```

Comment:
I am going to submit a separate PR to add an API to Source.toInstance that accepts a default instance when the value was filtered.

Reply:
https://github.com/spring-projects/spring-boot/pull/31323